includes/Post_Type_Base.php

```diff
@@ -294,12 +294,15 @@ public function get_parent_controller(): WP_REST_Controller {
 	 *
 	 * @since 1.14.0
 	 *
+	 * @SuppressWarnings(PHPMD.BooleanArgumentFlag)
```

Comment:
This is a legit warning for a legit code small

> A boolean flag argument is a reliable indicator for a violation of the Single Responsibility Principle (SRP). You can fix this problem by extracting the logic in the boolean flag into its own class or method.

A separate `get_default_archive_link` method would be less ambiguous and easier to comprehend.

Reply:
But that would mean having to method with basically the same logic.

Reply:
Shared logic could live in a shared private method if needed.

Reply:
I am not sure how that would look and what the point of it would be. I have updated the PR with some cleaner logic. https://github.com/GoogleForCreators/web-stories-wp/pull/10414/commits/3d17e603c8083d1f074a78b3249f6f63f99472bf

Reply:
Just making the default value `null` instead of `false` (for something that still is a boolean flag) is just way to trick the linter, but the issue still exists.

Reply:
I just don't see the point of having duplicated code here.

Reply:
I made a bash and breaking into two methods. See https://github.com/GoogleForCreators/web-stories-wp/pull/10414/commits/a44023e92f6cca1666e47c1100ce4544933e87de

Reply:
You could have copied over the tests and the improved PHPDoc as well.

Reply:
This code is still not ideal, it's just moving the `true === $post_type_obj->has_archive || $ignore_has_archive` check you had originally to a separate method.

Given that, when comparing this change with your original method, the original one would have been preferable over this.

Reply:
Have lots of lines of duplicated code was my issue with your PR.

What should I do here, revert back to the way it was?

Reply:
First of all let's add the tests that I've written so that I didn't write them for nothing.

Regarding the implementation, I would have preferred at least discussing my PR to avoid all this back and forth.

I _know_ my suggested code adds _some_ code duplication but it's an acceptable amount of duplicated code in favor of clearer separation of responsibilities.

That said, I don't have a super strong opinion on it though and it's not really worth arguing over, so please just revert your later changes

Reply:
Can you either revert your `get_archive_link` implementation to what you had originally (with suppressing the boolean flag warning) or implement my suggested changes?
I won't approve the current version.

Reply:
Reverted. But I kept the tests, as they as still valid.